src/Microsoft.ML.Data/DataView/DataViewConstructionUtils.cs

```diff
@@ -71,6 +71,13 @@ internal static SchemaDefinition GetSchemaDefinition<TRow>(IHostEnvironment env,
                 var schemaDefinitionCol = schemaDefinition.FirstOrDefault(c => c.ColumnName == name);
                 if (schemaDefinitionCol == null)
                     throw env.Except($"Type should contain a member named {name}");
+
+                //Always use column type from model as this type can be more specific.
+                //This can be corner case:
+                //For example, we can load an model whose schema contains Vector<Single, 38>
```

Comment:
Will there be a check or at least logging for this corner case? Maybe we should put in a check for arrays of size 0, which is what `float[]` would be. Also is there a check for the size of provided arrays matching the size of the corresponding `Vector`?

Reply:
test LoadModelWithOptionalColumnTransform is to test this case. 
for length check, it is applied at below:
[https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.Data/Scorers/SchemaBindablePredictorWrapper.cs#L129-L147](https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.Data/Scorers/SchemaBindablePredictorWrapper.cs#L129-L147)

---
In reply to: [439585282](https://github.com/dotnet/machinelearning/pull/5228#discussion_r439585282) [](ancestors = 439585282)